mmdeploy/codebase/mmseg/models/utils/up_conv_block.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+import torch
+
+from mmdeploy.core import FUNCTION_REWRITER
+from mmdeploy.utils import is_dynamic_shape
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmseg.models.utils.UpConvBlock.forward')
+def up_conv_block__forward(ctx, self, skip, x):
+    """Rewrite `forward` for default backend.
+
+    To support dynamic shape for UNet backbone,
+    upsample feature maps with `size` instead of `scale_factor`
+
+    Args:
+        ctx (ContextCaller): The context with additional information.
+        self: The instance of the original class.
+        skip (Tensor): Skip branch feature.
+        x (Tensor): Input feature to be upsampled.
+
+    Returns:
+        Tensor: Upsampled output feature map.
+    """
+    if is_dynamic_shape(ctx.cfg):
+        # upsample with `size` instead of `scale_factor`
+        from mmseg.ops import Upsample
+        for c in self.upsample.interp_upsample:
```

Comment:
upsample cound be a `ConvModule`.https://github.com/open-mmlab/mmsegmentation/blob/98984bf8553a83bc4a2231127d4da9323ac4db27/mmseg/models/utils/up_conv_block.py#L85

Is it safe to loop over upsample.interp_upsample?

Reply:
maybe we could check type of self.upsample at first.